src/persist-client/src/read.rs

```diff
@@ -194,7 +287,34 @@ where
         as_of: Antichain<T>,
         num_splits: NonZeroUsize,
     ) -> Result<Result<Vec<SnapshotSplit>, InvalidUsage>, LocationError> {
-        todo!("{:?}{:?}{:?}", timeout, as_of, num_splits);
+        trace!(
+            "ReadHandle::snapshot timeout={:?} as_of={:?} num_splits={:?}",
+            timeout,
+            as_of,
+            num_splits
+        );
+        let state = self.state.clone().into_inner();
+        let state = state.lock().await;
+        // TODO: This needs to attach a clone of the read capability to each
+        // part.
```

Comment:
Not in the supposedly new world where readers don't block compaction but there is a global `allow_compaction()` again, right?